server/novaboot-shell

```diff
@@ -124,6 +126,22 @@ run_subcommand() {
 	    mkdir -p "$HOME/tftproot"
 	    cd "$HOME/tftproot"
 	    exec "$@";;
+	"rsync-nfsroot "*" --server "*" . .")
+	    shift 1
+	    mkdir -p "$HOME/nfsroot"
+	    cd "$HOME/nfsroot"
+	    exec rsync "$@";;
+	"unfsd" *)
+	    rootdir="$2"
+	    shift 2
+
+	    [ -n "$board_ip" ] || die 'Missing configuration option $board_ip'
+	    echo "$HOME/nfsroot/$rootdir $board_ip(rw,no_root_squash)" > "$HOME/export"
+
+	    # start or restart unfs3 to reread updated `exports` file
+	    [ -n "$nfs_port" ] || die 'Missing configuration option $nfs_port'
+	    systemctl --user restart "unfsd@$nfs_port.service"
```

Comment:
The `@` unit names are needed when multiple instances of one service need to execute at the same time. Here, at most one unfsd server will be executed. Maybe, we can distinguish different novaboot users `unfsd@$NB_USER`, but even this is not necessary.

I think that the `unfsd` subcommand could just create a symlink (e.g. `$HOME/nfsroot-link`) pointing to `$HOME/nfsroot/$NB_USER` and and the `unfsd.service` and even the `export` file could just refer to this path.

Reply:
I'm using it for the port variable, what would another option would be to replace the command with a script which sources the `.novaboot-shell` config file.

Reply:
You can use `EnvironmentFile=...` - see systemd.exec(5). Some (randomly found) example is at https://coreos.com/os/docs/latest/using-environment-variables-in-systemd-units.html#environmentfile-directive